application/src/main/java/org/opentripplanner/updater/siri/SiriAlertsUpdateHandler.java

```diff
@@ -135,9 +135,9 @@ private TransitAlert mapSituationToAlert(
     TransitAlertBuilder alert = createAlertWithTexts(situation);
 
     if (
-      (alert.headerText() == null || alert.headerText().toString().isEmpty()) &&
-      (alert.descriptionText() == null || alert.descriptionText().toString().isEmpty()) &&
-      (alert.detailText() == null || alert.detailText().toString().isEmpty())
+      I18NString.hasNoValue(alert.headerText()) &&
+      I18NString.hasNoValue(alert.descriptionText()) &&
+      I18NString.hasNoValue(alert.detailText())
```

Comment:
Should this check not be part of `TransitAlertBuilder`?

Reply:
It should and I first put it in the builder - but the builder should not have the validation logic, the main class should. The problem is that if we validate when we construct the new entity, I would have to change the `SiriAlertsUpdateHandler` more than I like for a simple bug fix.